Archivist.lua

```diff
@@ -255,6 +261,33 @@ function proto:RegisterStoreType(prototype)
 	end
 end
 
+-- produces storable data from image
+function proto:Wind(storeType, image)
+	do -- arg validation
+		self:Assert(type(storeType) == "string" and self.prototypes[storeType], "Store type must be registered before loading data.")
+		self:Assert(image ~= nil, "Cannot wind nil data")
+	end
+
+	if self.prototypes[storeType].Wind then
+		return self.prototypes[storeType]:Wind(image)
+	else
+		return self:Archive(image)
+	end
+end
+
+function proto:Unwind(storeType, woundImage)
+	do -- arg validation
+		self:Assert(type(storeType) == "string" and self.prototypes[storeType], "Store type must be registered before loading data.")
+		self:Assert(woundImage ~= nil, "Cannot unwind nil data")
```

Comment:
Is there a world where possibly some custom Wind/Unwind implementation may actually _want_ to represent an empty image as `nil`? Wondering if it's worth relaxing this check in both operations - unless it causes actual issues elsewhere?

Reply:
you're correct to question this, I wrote the nil checks instinctively without actually ensuring that allowing nil would cause errors. 
There might be some subtle footgun, but considering the nature of this service I think it's fine to just let the consumer shoot themselves and figure it out.

Reply:
> Is there a world where possibly some custom Wind/Unwind implementation may actually want to represent an empty image as nil?
an implementation where the stored data was purely default implementation comes to mind, in which case it's possible to completely reconstruct it, and skip the space+computation costs.